### ISSUE TITLE
ConflictsWith error when creating a new project

### ISSUE THREAD

**Issue by lukassup** [terraform-google-modules/terraform-google-project-factory#675]
### TL;DR

Creating a project throws `Error: ConflictsWith` with `hashicorp/google v4.6.0` provider because it does not accept BOTH `org_id = ""` and `folder_id = "folders/1234567890"`. Looks like a bug introduced by a new `google` provider version.

### Expected behavior

Should create a project with no errors.

### Observed behavior

Throws error:
```
% terraform plan
╷
│ Error: ConflictsWith
│ 
│   with google_project.project,
│   on main.tf line 31, in resource "google_project" "project":
│   31:   org_id              = local.parent_type == "organizations" ? local.parent_id : ""
│ 
│ "org_id": conflicts with folder_id
╵
╷
│ Error: ConflictsWith
│ 
│   with google_project.project,
│   on main.tf line 32, in resource "google_project" "project":
│   32:   folder_id           = local.parent_type == "folders" ? local.parent_id : ""
│ 
│ "folder_id": conflicts with org_id
```

### Terraform Configuration

```hcl
module "project" {
  source = "terraform-google-modules/project-factory/google//modules/fabric-project"
  version = "11.3.0"
  name = "demo"
  prefix = "test"
  billing_account     = "xxxxxx-xxxxxx-xxxxxx"
  parent           = "folders/xxxxxxxxxxxx"
}
```

Smallest reproducible example (even without using this module)

```hcl
resource "google_project" "project" {
  org_id              = ""
  folder_id           = "folders/xxxxxxxxxxxx"
  project_id          = "demo-xxxxxx"
  name                = "demo"
  billing_account     = "xxxxxx-xxxxxx-xxxxxx"
}
```

### Terraform Version

```sh
% terraform version
Terraform v1.1.0
on darwin_amd64

provider: hashicorp/google 4.6.0
```

### Additional information

[`defaults()`](https://www.terraform.io/language/functions/defaults) is available since Terraform v0.14, looks like a potential fix:

```
org_id = defaults(EXPRESSION, null)
folder_id = defaults(EXPRESSION, null)
```

or should be able to add a version constraint in `required_providers` to the `google` provider version that does not have this bug

**Comment by bharathkkb** [comment]
Thanks for the report @lukassup 
In `fabric-project` we can use the same approach as main module where we set either field to null depending on inputs. Happy to review a PR for this.
https://github.com/terraform-google-modules/terraform-google-project-factory/blob/21871b2a147d654f6d59f868fd1a3fd5ad33c84b/modules/core_project_factory/main.tf#L30-L31

**Comment by lukassup** [comment]
PR ➡️ #676
